remote/remote_test.go

```diff
@@ -64,46 +53,108 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
-				defer h.DockerRmi(dockerClient, img.Name())
-				inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), img.Name())
+
+				os, err := img.OS()
+				h.AssertNil(t, err)
+				h.AssertEq(t, os, "linux")
+
+				osVersion, err := img.OSVersion()
 				h.AssertNil(t, err)
-				h.AssertEq(t, inspect.Os, "linux")
-				h.AssertEq(t, inspect.Architecture, "amd64")
-				h.AssertEq(t, inspect.RootFS.Type, "layers")
+				h.AssertEq(t, osVersion, "")
+
+				arch, err := img.Architecture()
+				h.AssertNil(t, err)
+				h.AssertEq(t, arch, "amd64")
 			})
 		})
 
 		when("#FromRemoteBaseImage", func() {
 			when("base image exists", func() {
-				var (
-					baseName         = "busybox"
-					err              error
-					existingLayerSha string
-				)
+				it("sets the initial state from a linux/arm base image", func() {
+					baseImageName := "arm64v8/busybox@sha256:50edf1d080946c6a76989d1c3b0e753b62f7d9b5f5e66e88bef23ebbd1e9709c"
+					existingLayerSha := "sha256:5a0b973aa300cd2650869fd76d8546b361fcd6dfc77bd37b9d4f082cca9874e4"
+
+					img, err := remote.NewImage(
+						repoName,
+						authn.DefaultKeychain,
+						remote.FromBaseImage(baseImageName),
+					)
+					h.AssertNil(t, err)
+
+					os, err := img.OS()
+					h.AssertNil(t, err)
+					h.AssertEq(t, os, "linux")
 
-				it.Before(func() {
-					err = h.PullImage(dockerClient, baseName)
+					osVersion, err := img.OSVersion()
 					h.AssertNil(t, err)
+					h.AssertEq(t, osVersion, "")
 
-					inspect, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), baseName)
+					arch, err := img.Architecture()
 					h.AssertNil(t, err)
+					h.AssertEq(t, arch, "arm64")
 
-					existingLayerSha = inspect.RootFS.Layers[0]
+					readCloser, err := img.GetLayer(existingLayerSha)
+					h.AssertNil(t, err)
+					defer readCloser.Close()
 				})
 
-				it("sets the initial state to match the base image", func() {
+				it("sets the initial state from a windows/amd64 base image", func() {
+					baseImageName := "mcr.microsoft.com/windows/nanoserver@sha256:06281772b6a561411d4b338820d94ab1028fdeb076c85350bbc01e80c4bfa2b4"
+					existingLayerSha := "sha256:26fd2d9d4c64a4f965bbc77939a454a31b607470f430b5d69fc21ded301fa55e"
+
 					img, err := remote.NewImage(
 						repoName,
 						authn.DefaultKeychain,
-						remote.FromBaseImage(baseName),
+						remote.FromBaseImage(baseImageName),
 					)
 					h.AssertNil(t, err)
 
+					os, err := img.OS()
+					h.AssertNil(t, err)
+					h.AssertEq(t, os, "windows")
+
+					osVersion, err := img.OSVersion()
+					h.AssertNil(t, err)
+					h.AssertEq(t, osVersion, "10.0.17763.1040")
+
+					arch, err := img.Architecture()
+					h.AssertNil(t, err)
+					h.AssertEq(t, arch, "amd64")
+
 					readCloser, err := img.GetLayer(existingLayerSha)
 					h.AssertNil(t, err)
 					defer readCloser.Close()
 				})
+
+				when("base image is a multi-OS/Arch manifest list", func() {
+					it("returns a base image matching the runtime GOOS/GOARCH", func() {
```

Comment:
Should this be changed to say "defaults to linux"?

Reply:
Agree, but I am going to merge to avoid delaying this large change any longer than necessary. We can also PR more test cleanup.